File: 20220928/모범답안/20220928_08.js/CountTruncateLetterArray_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.medium.CountTruncateLetterArray.CountTruncateLetterArray import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        arr = ['no pain', 'no', 'gain']

        # when
        result = solution(arr)

        # then
        self.assertEqual(3, result)

    def test_2(self):
        # given
        arr = ['Hello', 'world', 'Nice World']

        # when
        result = solution(arr)

        # then
        self.assertEqual(4, result)

    def test_3(self):
        # given
        arr = ['We can give advice but we cannot give conduct']

        # when
        result = solution(arr)

        # then
        self.assertEqual(8, result)

    def test_4(self):
        # given
        arr = ['Better late', 'than never']

        # when
        result = solution(arr)

        # then
        self.assertEqual(4, result)

    def test_5(self):
        # given
        arr = ['Slow', 'and', 'steady', 'win the race']

        # when
        result = solution(arr)

        # then
        self.assertEqual(6, result)

    def test_6(self):
        # given
        arr = ['Rome is', 'not built', 'in a day']

        # when
        result = solution(arr)

        # then
        self.assertEqual(7, result)


if __name__ == '__main__':
    unittest.main()

File: 20220928/모범답안/20220928_08.js/CountTruncateLetterArray_test.js
import assert from 'assert'
import solution from './CountTruncateLetterArray.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const arr = ['no pain', 'no', 'gain']

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(3, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const arr = ['Hello', 'world', 'Nice World']

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(4, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const arr = ['We can give advice but we cannot give conduct']

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(8, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const arr = ['Better late', 'than never']

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(4, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const arr = ['Slow', 'and', 'steady', 'win the race']

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(6, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const arr = ['Rome is', 'not built', 'in a day']

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(7, result)
    done();
  });
})
